provider/maas/devices.go

```diff
@@ -230,7 +230,11 @@ func (env *maasEnviron) deviceInterfaceInfo(deviceID instance.Id, nameToParentNa
 			}
 
 			nicInfo.CIDR = link.Subnet.CIDR
-			nicInfo.Address = corenetwork.NewProviderAddressInSpace(link.Subnet.Space, link.IPAddress)
+			// NOTE(achilleasa): the original code used a last-write-wins
```

Comment:
This might be a little misleading to someone unfamiliar reading it.
Maybe something like:
> This preserves the long-standing last-write-wins behaviour...